fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java

```diff
@@ -0,0 +1,110 @@
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.common.security.authentication;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class HadoopUGI {
+
+    private static final Logger LOG = LogManager.getLogger(HadoopUGI.class);
+    public static String HADOOP_USER_NAME = "hadoop.username";
+    public static String HADOOP_SECURITY_AUTHENTICATION = "hadoop.security.authentication";
+    public static String HADOOP_KERBEROS_PRINCIPAL = "hadoop.kerberos.principal";
+    public static String HADOOP_KERBEROS_AUTHORIZATION = "hadoop.security.authorization";
+    public static String HADOOP_KERBEROS_KEYTAB = "hadoop.kerberos.keytab";
+
+    /**
+     * login and return hadoop ugi
+     * @param conf config
+     * @return ugi
+     */
+    public static UserGroupInformation loginWithUGI(Configuration conf) {
+        UserGroupInformation ugi;
+        String authentication = conf.get(HADOOP_SECURITY_AUTHENTICATION, null);
+        if (AuthType.KERBEROS.getDesc().equals(authentication)) {
+            String principal = conf.get(HADOOP_KERBEROS_PRINCIPAL);
+            try {
+                //  login hadoop with keytab and try checking TGT
+                ugi = UserGroupInformation.getLoginUser();
+                LOG.debug("Current login user: {}", ugi.getUserName());
+                if (ugi.hasKerberosCredentials() && StringUtils.equals(ugi.getUserName(), principal)) {
+                    // if the current user is logged by kerberos and is the same user
+                    // just use checkTGTAndReloginFromKeytab because this method will only relogin
+                    // when the TGT is expired or is close to expiry
+                    ugi.checkTGTAndReloginFromKeytab();
+                    return ugi;
+                }
+            } catch (IOException e) {
+                LOG.warn("A SecurityException occurs with kerberos, do login immediately.", e);
+            }
+            String keytab = conf.get(HADOOP_KERBEROS_KEYTAB);
```

Comment:
Some place may required other principal and keytab, eg, hive metastore client.
HADOOP_KERBEROS_KEYTAB and HADOOP_KERBEROS_PRINCIPAL is only for hdfs

Reply:
OK, add hive krb properties